devicetypes/heltun/heltun-tps05-switch.src/heltun-tps05-switch.groovy

```diff
@@ -0,0 +1,667 @@
+	def map = [:]
+	def roomTemperature = 1
+	def himidity = 5
+	def illuminance = 3
+	def locaScale = getTemperatureScale() //HubScale
+	def deviceScale = (cmd.scale == 1) ? "F" : "C" //DeviceScale
+	def child = childDevices?.find {channelNumber(it.deviceNetworkId) == 1 }
+	if (roomTemperature == cmd.sensorType) {
+		def deviceTemp = cmd.scaledSensorValue
+		def scaledTemp = (deviceScale == locaScale) ? deviceTemp : (deviceScale == "F" ? roundC(fahrenheitToCelsius(deviceTemp)) : celsiusToFahrenheit(deviceTemp).toDouble().round(0).toInteger())
+		map.name = "temperature"
+		map.value = scaledTemp
+		map.unit = locaScale
+		sendEvent(map)
+	} else if (himidity == cmd.sensorType) {
+		map.name = "humidity"
+		map.value = cmd.scaledSensorValue.toInteger()
+		map.unit = "%"
+		sendEvent(map)
+	} else if (illuminance == cmd.sensorType) {
+		map.name = "illuminance"
+		map.value = cmd.scaledSensorValue
+		sendEvent(map)
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.configurationv2.ConfigurationReport cmd) {
+	def parameter = parameterMap().find( {it.paramNum == cmd.parameterNumber } ).name
+	if (state."$parameter".value == cmd.scaledConfigurationValue){
+		state."$parameter".state = "configured"
+	}
+	else {
+		state."$parameter".state = "error"
+	}
+	configParam()
+}
+
+def updated() {
+	if (childDevices && device.label != state.oldLabel) {
+		childDevices.each {
+			def newLabel = getChildName(channelNumber(it.deviceNetworkId))
+			it.setLabel(newLabel)
+		}
+		state.oldLabel = device.label
+	}
+	initialize()
+}
+
+def initialize() {
+	runIn(3, "checkParam")
+}
+
+def installed() {
+	def numberOfButtons = 5
+	state.numberOfButtons = numberOfButtons
+	state.oldLabel = device.label
+	def existingChildren = getChildDevices()
+	for (i in 1..numberOfButtons) {
+		def ButtonNetworkId = "${device.deviceNetworkId}:${i+2*numberOfButtons}"
+		def RelayNetworkId = "${device.deviceNetworkId}:${i+numberOfButtons}"
+		def BacklightNetworkId = "${device.deviceNetworkId}:${i}"
+		def childRelayExists = (existingChildren.find {child -> child.getDeviceNetworkId() == RelayNetworkId} != NULL)
+		def childButtonExists = (existingChildren.find {child -> child.getDeviceNetworkId() == ButtonNetworkId} != NULL)
+		def childBacklightExists = (existingChildren.find {child -> child.getDeviceNetworkId() == BacklightNetworkId} != NULL)
+		if (!childBacklightExists ) {
+			def child = addChildDevice("HE-BACKLIGHT", BacklightNetworkId, device.hubId, [completedSetup: true, label: getChildName(i), isComponent: false])
+		}
+		if (!childRelayExists) {
+			addChildDevice("HE-RELAY", RelayNetworkId, device.hubId,[completedSetup: true, label: getChildName(i+numberOfButtons), isComponent: false])
+		}
+		if (!childButtonExists ) {
+			def child = addChildDevice("HE-BUTTON", ButtonNetworkId, device.hubId, [completedSetup: true, label: getChildName(i+2*numberOfButtons), isComponent: true, componentName: "button$i", componentLabel: "Button ${i}"])
```

Comment:
You forgot to include childs' DTHes in the PR.

Reply:
Just added them

Reply:
Still not seeing any in this PR.

Reply:
I was not able to do changes to this PR.
I made a new PR, you can find it at https://github.com/Sarkis008/SmartThingsPublic/tree/master/devicetypes/heltun
Can we continue there?

Reply:
Link you've provided is not a PR.

Reply:
I have no experience with Github.
Could you please let me know on which file I need to do the changes ?

Reply:
You need to add[heltun-child-backlight.src](https://github.com/Sarkis008/SmartThingsPublic/tree/master/devicetypes/heltun/heltun-child-backlight.src) and [heltun-child-relay.src](https://github.com/Sarkis008/SmartThingsPublic/tree/master/devicetypes/heltun/heltun-child-relay.src) which, I can see, are available at `master` branch of your repository, to the branch connected to this PR. What is that branch, you can find out just beneath the title of the PR.
![image](https://user-images.githubusercontent.com/41617389/154054731-523d5971-f52b-4533-b0f5-fc7077c2b401.png)

Reply:
@Sarkis008 unfortunately, I'm not familiar with how did you submit through Developer Workspace and even how it creates PR per request. I can only suggest contacting helpdesk (afaik you have a conversation ongoing) or just cancel request for this device, and then create another one, but in similar way as you did for the device in https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/77748, so you'd have access to source branch.

Reply:
Or just create a new, separate PR without canceling your Developer Workspace request.

Reply:
@PKacprowiczS it seems like I created a new PR ))
Could you please check this https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/77846